classes/user/Blacklist.php

```diff
@@ -0,0 +1,115 @@
+<?php
+class Blacklist {
+    public function __construct(
+        public System $system,
+        public int $user_id,
+        public string $encodedBlacklist = "[]",
+        public array $blacklist = array(),
+        public bool $update = false
+    ) {
+        $this->loadBlacklist();
+    }
```

Comment:
If the class needs DB info to initialize, I kind of like the pattern to make a static function to do that like `Blacklist::fromUserId`, and make the constructor not trigger any DB queries. This has the advantage that if you want to optimize later and do a batch query for a bunch of blacklists, you can just pass each row of blacklist data into the constructor

Reply:
Will include this with the Inbox update